Core/source/core/pgp-key.ts

```diff
@@ -135,6 +135,13 @@ export class PgpKey {
     return p instanceof SecretKeyPacket || p instanceof SecretSubkeyPacket;
   }
 
+  public static validateAllDecryptedPackets = async (key: Key): Promise<void> => {
+    const packets = key.toPacketList() as PacketList<SecretKeyPacket>;
+    for (const prvPacket of packets.filter(PgpKey.isPacketPrivate).filter(packet => packet.isDecrypted())) {
+      await (prvPacket as SecretKeyPacket).validate(); // gnu-dummy never raises an exception, invalid keys raise exceptions
```

Comment:
should be type cast to `BaseSecretKeyPacket`

Reply:
update, thanks!

Reply:
sir, I made a recent change that would work without further typecasting but by checking if the prvPacket is an instance of SecretKeyPacket. would that be ok?